File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    target = 1
    count = 1
    duplication_check = []
    
    for word in words:
        # 중복 단어인지 체크
        if word in duplication_check:
            return [target, count]   
        
        # 끝말잇기가 적용되었는지 체크
        if len(duplication_check) >= 1:
            if duplication_check[-1][-1] != word[0]:
                return [target, count]

        duplication_check.append(word)
        
        target += 1
        if target == n + 1:
            target = 1
            count += 1

    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    var answer = [];
    const ref = new Map();
    let count = 0;
    
    for (let i = 0; i < words.length; i++) {
        count = i;
        const curr = words[i];
        // 3번 체크
        if (i !== 0) {
            const prev = words[i - 1];
            if (prev[prev.length - 1] !== curr[0]) {
                break;
            }
        }
        
        // 4번 체크
        if (ref.has(curr)) break;
        
        ref.set(curr, true);
    }
    
    return ref.size === words.length ? [0, 0] : [(count % n) + 1, parseInt(count / n) + 1];
}